Managers/PrimitiveMan.cpp

```diff
@@ -0,0 +1,230 @@
+			Vector drawStart = m_StartPos - targetPos;
+			circle(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			circle(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
+			circle(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::CircleFillPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			circlefill(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			circlefill(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
+			circlefill(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::TextPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
+
+			if (m_IsSmall) {
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
+			} else {
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
+			}
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
+			if (m_IsSmall) {
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
+			} else {
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::BitmapPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!m_pBitmap) {
+			return;
+		}
+		Vector pos[2];
+		unsigned short passes = 0;
+
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
```

Comment:
Can you rework this method a little so it's consistent with the rest of them and not stupid? There's a lot of bad stuff here
1. Variable cleanup - passes isn't used anywhere, delete it. Also delete pos, it'll be unneeded junk.
2. Reorganization and consistency - like the rest of them, make this call the allegro draw method in multiple places. Look at drawCircle for an example of what this **should** look like (but with `pivot_scaled_sprite` instead of `circle` method).